yt_dlp/extractor/generic.py

```diff
@@ -2184,12 +2186,30 @@ def report_detected(self, name, num=1, note=None):
 
         self._downloader.write_debug(f'Identified {num} {name}{format_field(note, None, "; %s")}')
 
-    def _fragment_query(self, url):
+    def _extra_manifest_info(self, info, manifest_url, is_info_dict=True):
+        extra_info = {}
+
         if self._configuration_arg('fragment_query'):
-            query_string = urllib.parse.urlparse(url).query
+            query_string = urllib.parse.urlparse(manifest_url).query
             if query_string:
-                return {'extra_param_to_segment_url': query_string}
-        return {}
+                extra_info['extra_param_to_segment_url'] = query_string
```

Comment:
```suggestion
                info['extra_param_to_segment_url'] = query_string
```
Why have an `extra_info`?

Reply:
good point